### ISSUE TITLE
Strange behavior of the auto-correction for `Layout/TrailingWhitespace`

### ISSUE THREAD

**Issue by khiav223577** [rubocop-hq/rubocop#9392]
## Expected behavior

Expect trailing whitespaces to be removed.

## Actual behavior

It is replaced with `#{' '}`

## Steps to reproduce the problem

Add example.rb:

### example.rb
```rb
class Example
end

Example.module_eval(<<-CODE, __FILE__, __LINE__ + 1)
  def self.test
    p 'test' 
  end
CODE

Example.test
```

Run auto-correcting:
```bash
$ rubocop example.rb --auto-correct --only Layout/TrailingWhitespace
```

### example.rb (auto-corrected)
```rb
class Example
end

Example.module_eval(<<-CODE, __FILE__, __LINE__ + 1)
  def self.test
    p 'test'#{' '}
  end
CODE

Example.test
```

## RuboCop version

```
$ rubocop -V
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.8-compliant syntax, but you are running 2.5.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
1.8.1 (using Parser 3.0.0.0, rubocop-ast 1.3.0, running on ruby 2.5.5 x86_64-linux)
```

**Comment by dvandersluis** [comment]
The cop adds that on purpose in heredocs (see #8914). @marcandre might have more insight as to why?

**Comment by marcandre** [comment]
Indeed, this is by design.

The idea is that this Cop is safe and (thus will produce auto-correct without introducing a bug).

There is no way for RuboCop to know if you intended the trailing whitespace in the heredoc or not; the cop assume you did, and the correction makes it possible to see it. You are of course welcome to modify it and remove it altogether.

It also makes it easier for people like me that have their editors set to automatically remove trailing whitespace 😆
